db/src/db.rs

```diff
@@ -387,21 +388,30 @@ impl TypedSQLValue for TypedValue {
             &TypedValue::Long(x) => (Value::Integer(x), ValueType::Long),
             &TypedValue::Double(x) => (Value::Float(x), ValueType::Double),
             &TypedValue::String(ref x) => (Value::Text(x.clone()), ValueType::String),
-            &TypedValue::Keyword(ref x) => (Value::NamespacedKeyword(to_namespaced_keyword(&x).unwrap()), ValueType::Keyword),
+            &TypedValue::Keyword(ref x) => {
+                match to_namespaced_keyword(&x) {
+                    Some(x) => (Value::NamespacedKeyword(x), ValueType::Keyword),
+                    // TODO Use custom ErrorKind https://github.com/brson/error-chain/issues/117
+                    None => panic!(ErrorKind::NotYetImplemented(format!("InvalidNamespacedKeyword: {}", x))),
+                }
+            },
         }
     }
 }
 
 /// Read the ident map materialized view from the given SQL store.
 pub fn read_ident_map(conn: &rusqlite::Connection) -> Result<IdentMap> {
     let mut stmt: rusqlite::Statement = conn.prepare("SELECT ident, entid FROM idents")?;
-    let m = stmt.query_and_then(&[], |row| -> Result<(String, Entid)> {
-        Ok((row.get(0), row.get(1)))
+    let m = stmt.query_and_then(&[], |row| -> Result<(symbols::NamespacedKeyword, Entid)> {
+        let ident: String = row.get(0);
+        to_namespaced_keyword(&ident)
```

Comment:
It's possible to make `let ident: NamespacedKeyword = row.get(0)` just work, by implementing `rusqlite::types::FromSql`.  However, that's awkward because the type is in the `edn` crate and the trait is in the `rusqlite` trait.  It might be worth newtyping this to be automatic, but we can do that as follow-up.  This might just get better with your next ticket, too.